jgrapht-core/src/main/java/org/jgrapht/alg/cycle/ChordalityInspector.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * (C) Copyright 2018, by Timofey Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.cycle;
+
+import org.jgrapht.Graph;
+import org.jgrapht.Graphs;
+import org.jgrapht.graph.AsUndirectedGraph;
+import org.jgrapht.traverse.GraphIterator;
+import org.jgrapht.traverse.LexBreadthFirstIterator;
+import org.jgrapht.traverse.MaximumCardinalityIterator;
+
+import java.util.*;
+
+/**
+ * Allows testing chordality of a graph. The inspected {@code graph} is specified at construction time
+ * and cannot be modified. Currently chordality of a graph is tested via {@link MaximumCardinalityIterator}
+ * by default. When {@link IterationOrder#LEX_BFS} is specified as a second constructor parameter, this
+ * {@code ChordalityInspector} uses {@link LexBreadthFirstIterator} to compute perfect elimination order.
+ * <p>
+ * A <a href="https://en.wikipedia.org/wiki/Chordal_graph">chordal graph</a> is one in which all cycles of
+ * four or more vertices have a chord, which is an edge that is not part of the cycle but connects two vertices
+ * of the cycle.
+ * <p>
+ * A graph is chordal iff its the vertices can be arranged into a perfect elimination order. Perfect elimination
```

Comment:
I think you mean "More than one perfect elimination order may exist for a given graph.  Either maximum cardinality search or lexicographical breadth-first search can be used to produce such an order."